src/formatter-settings/index.ts

```diff
@@ -216,13 +235,13 @@ export class JavaFormatterSettingsEditorProvider implements vscode.CustomTextEdi
     }
 
     private async inheritVSCodeSettings(webviewPanel: vscode.WebviewPanel) {
-        const javaTabSize: number | undefined = await this.findJavaSpecificSetting<number>("editor.tabSize");
-        const javaSpacePolicy: boolean | undefined = await this.findJavaSpecificSetting<boolean>("editor.insertSpaces");
-        const vscodeTabSize: number | undefined = vscode.workspace.getConfiguration().get<number>("editor.tabSize");
-        const vscodeSpacePolicy: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>("editor.insertSpaces");
-        const tabSize = (javaTabSize === undefined) ? vscodeTabSize : javaTabSize;
-        const spacePolicy = (javaSpacePolicy === undefined) ? vscodeSpacePolicy : javaSpacePolicy;
-        for (const setting of initializeSupportedVSCodeSettings()) {
+        const tabSize: number | undefined = await this.getSetting<number>("editor.tabSize");
+        const spacePolicy: boolean | undefined = await this.getSetting<boolean>("editor.insertSpaces");
+        if (tabSize === undefined || spacePolicy === undefined) {
```

Comment:
Give them a default value instead of undefined. Meanwhile, log the message when undefined happens